addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -512,9 +638,21 @@ private void updateChannelStates(LxControl control) {
             if (value != null && value >= 0 && value < LxControlLightController.NUM_OF_SCENES) {
                 updateState(channelId, new DecimalType(value));
             }
-            if (controller.sceneNamesUpdated()) {
-                setLightControllerStateDescription(channelId, controller);
+        } else if (control instanceof LxControlLightControllerV2) {
+            LxControlLightControllerV2 controller = (LxControlLightControllerV2) control;
+            List<Integer> activeMoods = controller.getActiveMoods();
+            // update the single mood channel state
+            if (activeMoods.size() == 1) {
+                updateState(channelId, new DecimalType(activeMoods.get(0)));
+            } else {
+                updateState(channelId, UnDefType.UNDEF);
             }
+            // update the individual mood mixing channels
+            Map<LxUuid, LxControlMood> allMoods = controller.getMoods();
+            allMoods.values().forEach(v -> {
+                // we update moods like all other switches with no special dedicated code
+                updateChannelStates(getChannelIdForControl(v, 0), v);
+            });
```

Comment:
I agree, can I submit a separate PR for this, as this will be some change to the way the classes are designed. I'd love to get the new controller functionality committed before I refactor that.

Reply:
👍